src/Grid/utils.js

```diff
@@ -1,29 +1,43 @@
+import PropTypes from 'prop-types';
+import breakpoints from '../breakpoints.json';
+
+const bps = Object.keys(breakpoints);
+export const bpPropClass = (bp, prop) => `${prop}On${bp.substr(0, 1).toUpperCase() + bp.substr(1)}`;
+
 /**
- * Removes properties from the given object.
- * This method is used for removing valid attributes from component props prior to rendering.
+ * Generate string with classnames from props names and breakpoints
  *
- * @param {Object} object
- * @param {Array} remove
- * @returns {Object}
+ * @param {Object} props
+ * @param {String Array} numericProps
+ * @param {String Array} booleanProps
+ * @param {Boolean} hasBreakpoint
+ * @returns {String}
  */
-export function removeProps(object, remove) {
-  const result = {};
-
-  for (const property in object) {
-    if (object.hasOwnProperty(property) && remove.indexOf(property) === -1) {
-      result[property] = object[property];
-    }
-  }
-
-  return result;
-}
+export const generateClassnames = (props, numericProps, booleanProps, hasBreakpoint = false) =>
```

Comment:
Improve code readability by creating small functions for each array in the `#concat ` method. Ask your self, what does this piece of code?

```js
numericProps.reduce((names, prop) => {
      const propValue = props[bpPropClass(bp, prop)];
      return names.concat(propValue ? ` ${bp}-${prop}-${propValue}` : '');
    }, '')
```

Better, avoid value declaration on iteration callbacks and embrace functional JS

```js
const getClassName = bp => (
  hasBreakpoint && props[bp] ? ` ${bp}-${props[bp]}` : null
);

const getNumericClassNames = bp =>
  numericProps
    .map(prop => ({
      name: prop,
      className: props[bpPropClass(bp, prop)]
    }))
    .filter(bpObj => props[bpObj.className])
    .map(bpObj => `${bp}-${bpObj.name}-${bpObj.className}`);

const getBooleanClassNames = bp =>
  booleanProps
    .filter(
      prop => (hasBreakpoint ? props[bpPropClass(bp, prop)] : props[prop])
    )
    .map(prop => (hasBreakpoint ? `${bp}-${prop}` : prop));

bps
  .map(bp => {
    [
      getClassName(bp),
      getNumericClassNames(bp),
      getBooleanClassNames(bp)
    ];
  })
  .join('');
```

The same approach can be used on `generatePropTypes` method.